WebViewControl/ResourceUrl.cs

```diff
@@ -56,33 +59,46 @@ public partial class ResourceUrl {
+        /// embedded://webview/AssemblyName-AssemblyVersion/Path/To/Resource
         /// </summary>
         internal static string[] GetEmbeddedResourcePath(Uri resourceUrl) {
             if (ContainsAssemblyLocation(resourceUrl)) {
                 var indexOfPath = resourceUrl.AbsolutePath.IndexOf(AssemblyPathSeparator);
                 return resourceUrl.AbsolutePath.Substring(indexOfPath + 1).Split(new [] { PathSeparator }, StringSplitOptions.None);
             }
             var uriParts = resourceUrl.Segments;
-            return uriParts.Skip(1).Select(p => p.Replace(PathSeparator, "")).ToArray();
+            var assemblyName = uriParts[1].Split(AssemblyVersionSeparator).First();
+            return uriParts.Skip(2).Prepend(assemblyName).Select(p => p.Replace(PathSeparator, "")).ToArray();
         }
 
         /// <summary>
         /// Supported syntax:
         /// embedded://webview/assembly:AssemblyName;Path/To/Resource
-        /// embedded://webview/AssemblyName/Path/To/Resource (AssemblyName is also assumed as default namespace)
+        /// embedded://webview/assembly:AssemblyName-AssemblyVersion;Path/To/Resource
+        /// embedded://webview/AssemblyName/Path/To/Resource  (AssemblyName is also assumed as default namespace)
+        /// embedded://webview/AssemblyName-AssemblyVersion/Path/To/Resource
         /// </summary>
-        public static string GetEmbeddedResourceAssemblyName(Uri resourceUrl) {
+        public static (string, Version) GetEmbeddedResourceAssemblyNameAndVersion(Uri resourceUrl) {
             if (ContainsAssemblyLocation(resourceUrl)) {
                 var resourcePath = resourceUrl.AbsolutePath.Substring((PathSeparator + AssemblyPrefix).Length);
                 var indexOfPath = Math.Max(0, resourcePath.IndexOf(AssemblyPathSeparator));
-                return resourcePath.Substring(0, indexOfPath);
+                return GetAssemblyNameAndVersion(resourcePath.Substring(0, indexOfPath));
             }
             if (resourceUrl.Segments.Length > 1) {
                 var assemblySegment = resourceUrl.Segments[1];
-                return assemblySegment.EndsWith(PathSeparator) ? assemblySegment.Substring(0, assemblySegment.Length - PathSeparator.Length) : assemblySegment; // default assembly name to the first path
+                // default assembly name to the first path
+                return GetAssemblyNameAndVersion(assemblySegment.EndsWith(PathSeparator) ? assemblySegment.Substring(0, assemblySegment.Length - PathSeparator.Length) : assemblySegment);
             }
-            return string.Empty;
+            return (string.Empty, null);
+        }
+
+        private static (string, Version) GetAssemblyNameAndVersion(string assemblyNameAndVersion) {
+            var parts = assemblyNameAndVersion.Split(AssemblyVersionSeparator);
+            return parts.Length == 1 ?
+                (parts[0], null) :
+                (parts[0], new Version(parts[1]));
```

Comment:
a bit more robust, in case input is not well formed
```suggestion
            return parts.Length == 2 ?
                (parts[0], new Version(parts[1])) :
                assemblyNameAndVersion;
```